modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js

```diff
@@ -12,160 +12,275 @@
  * details.
  */
 
-import '../FieldBase/FieldBase.es';
+import './ImagePickerRegister.soy';
 
-import './ImagePickerAdapter.soy';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal, {useModal} from '@clayui/modal';
+import {ItemSelectorDialog, cancelDebounce, debounce} from 'frontend-js-web';
+import React, {useEffect, useRef, useState} from 'react';
 
-import './ImagePickerRegister.soy';
+import getConnectedReactComponentAdapter from '../util/ReactComponentAdapter.es';
+import {connectStore} from '../util/connectStore.es';
+import templates from './ImagePickerAdapter.soy';
+
+const useDebounceCallback = (callback, milliseconds) => {
+	const callbackRef = useRef(debounce(callback, milliseconds));
+
+	return [callbackRef.current, () => cancelDebounce(callbackRef.current)];
+};
+
+const ImagePicker = ({
+	id,
+	inputValue = '',
+	itemSelectorURL,
+	name,
+	onClearClick,
+	onDescriptionChange,
+	onFieldChanged,
+	portletNamespace,
+	readOnly,
+}) => {
+	const [imageValues, setImageValues] = useState({});
+	const [modalVisible, setModalVisible] = useState(false);
+
+	useEffect(() => {
+		setImageValues({
+			...{description: '', title: '', url: ''},
+			...JSON.parse(inputValue || '{}'),
+		});
+	}, [inputValue]);
 
-import './ReactImagePickerAdapter.es';
+	const {observer} = useModal({
+		onClose: () => setModalVisible(false),
+	});
 
-import Component from 'metal-component';
-import Soy from 'metal-soy';
-import {Config} from 'metal-state';
+	const _dispatchValue = ({clear, value}, callback) => {
```

Comment:
@diegonvs I think we no longer need the underscore as a default here.

Reply:
Yep, renaming! Thanks for point up!